### PR TITLE
Controller rspec

### PR DESCRIPTION
**Pull request by Josphat205**
### **In this Branch**

- Added RSpec as a gem to your project.
- Created a Request spec file for all your controllers.
**In each action**

- I tested  If the response status was correct.
- I tested If a correct template was rendered
- I tested If the response body includes the correct placeholder text.

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -2,6 +2,18 @@
...
-    pending "add some examples (or delete) #{__FILE__}"
+    it('status should be :ok') do
+      get '/users/index'
+      expect(response).to have_http_status(:ok)
+    end
+
+    it('rendered :index template') do
+      get '/users/index'
+      expect(response).to render_template(:show)
+    end
```
Comment (FrederickMih) [inline spec/requests/users_spec.rb]: - [x] @Josphat205 according to this test, user request for an index template as seen in the 'it' block but the show template(`render_template(:show`) is rendered. Let's kindly it.

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -2,6 +2,18 @@
 
 RSpec.describe 'Users', type: :request do
```
Comment (FrederickMih) [inline spec/requests/users_spec.rb:3]: - [ ] The index and show actions for the user controller should be tested as requested in the requirements. They should be a `describe` block for both `index` and `show` actions for best practices. **Hints**


```ruby
describe 'GET /index' do
   it('status should be :ok') do
      get '/users/index'
      expect(response).to have_http_status(:ok)
    end

    it('rendered :index template') do
      get '/users/index'
      expect(response).to render_template(:index)
    end

    it 'body should includes correct placeholder text' do
      get '/users/index'
      expect(response.body).to include('Here is all users')
    end
end

describe 'GET /show' do
  ...
end
```

- kindly do the same for GET/show request

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+
+    it('should render :show template') do
+      expect(response).to render_template(:show)
+    end
+
+    it 'body should includes correct placeholder text' do
+      expect(response.body).to include('Here is a single post')
+    end
   end
 end
```
Comment (FrederickMih) [inline spec/requests/posts_spec.rb]: - [ ] Please kindly check the comment above :arrow_up:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @Josphat205, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️

- [ ] Let's kindly update the date(year) and name of the license file. The date is obsolete.

_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/users_spec.rb (hunk 1 above)
Comment (Josphat205) [reply spec/requests/users_spec.rb]: updated

**Review by TadesseDev — CHANGES_REQUESTED** [review]
x

**Review by TadesseDev — APPROVED** [review]
### Hi @Josphat205 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)

## Highlights
 - [ ] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**